language/transaction-builder/src/lib.rs

```diff
@@ -564,3 +569,26 @@ encode_txn_script! {
     script: RemoveParentAccount,
     doc: "Removes the parent account at `parent_address`. It can be recertified in the future however."
 }
+
+//...........................................................................
+// WriteSets
+//...........................................................................
+
+pub fn encode_stdlib_upgrade_transaction(option: StdLibOptions) -> ChangeSet {
```

Comment:
Would it be possible to reuse the code that genesis uses to build this `ChangeSet` to ensure that the implementations don't diverge?

Reply:
The stdlib upgrade is only updating code, or also update resource? genesis code also generates the resource.

Reply:
This approach only updates code. But you raise an interesting point: it seems very important to check that data format of existing resources has not been changed.